ingest/expression_files/mtx.py

```diff
@@ -153,94 +152,60 @@ def extract_feature_barcode_matrices(self):
         ]
 
     def transform(self):
-        start_time = datetime.datetime.now()
         num_processed = 0
-        last_idx = 0
+        prev_idx = 0
         gene_models = []
         data_arrays = []
         exp_cells = []
         exp_scores = []
-        visited_expression_idx = [0]
-        model_id = None
 
         # All observed cells
-        data_arrays.extend(
-            GeneExpression.create_data_arrays(
-                name=f"{self.cluster_name} Cells",
-                array_type="cells",
-                values=self.cells,
-                linear_data_type="Study",
-                linear_data_id=self.study_file_id,
-                **self.data_array_kwargs,
-            )
-        )
+        for data_array in GeneExpression.create_data_arrays(
+            name=f"{self.cluster_name} Cells",
+            array_type="cells",
+            values=self.cells,
+            linear_data_type="Study",
+            linear_data_id=self.study_file_id,
+            **self.data_array_kwargs,
+        ):
+            data_arrays.append(data_array)
         for row in self.mtx_file:
             raw_gene_idx, raw_barcode_idx, raw_exp_score = row.split()
             current_idx = int(raw_gene_idx)
-            gene_id, gene = self.genes[current_idx - 1].split("\t")
-            if current_idx != last_idx:
-                is_sorted = MTXIngestor.is_sorted(current_idx, visited_expression_idx)
-                if not is_sorted:
+            if current_idx != prev_idx:
+                if not current_idx == prev_idx + 1:
```

Comment:
There is no guarantee that the index will increment sequentially like this.  There may not be any observed expression for a given gene.  So if we are still enforcing the sorting of `mtx` files (I thought we had a way to not do this?) then you just want to make sure the index isn't _less than_ the current index.  The values should only go up.

Reply:
7226968